### ISSUE TITLE
About `neg_metrics` -> `p_neg_weight`

### ISSUE THREAD

**Issue by filaPro** [strongwolf/DW#11]
Hi @strongwolf,

I have one more question about the code. Looks like in this [line](https://github.com/strongwolf/DW/blob/main/dw_head.py#L169-L170) you use non one-to-one correspondence between `gt_labels` and `num_classes`. So in case if ambiguous anchor some values in `neg_metrics` will be simply ignored. And what is more strange the ignored value depends on the order of labels in `gt_labels`. Like here the value `2` is ignored in the resulting tensor:
```
>>> t = torch.tensor([[1]])
>>> t[[0, 0], [0, 0]] = torch.tensor([2, 3])
>>> t
tensor([[3]])
```

Do I understand it correctly?

**Comment by strongwolf** [comment]
Yes, your understanding is correct.  This phenomenon only occurs when an anchor is inside two gt boxes that have the same category label. I have tried to use min() or max() to deal with this situation and found there is no much difference from the original implementation. The reason I guess is that the number of such ambiguous anchors is very small compared with the total number of anchors.

**Comment by filaPro** [comment]
I see, thank you.
